### PR TITLE
Introduce version 0x81, including microsecond timestamps

### PR DESCRIPTION
**Pull request by dolph**
This change introduces a second version of the specification (0x81), and asserts that timestamps in tokens versioned as 0x81 be evaluated as microseconds since the epoch. The timestamp is still a 64-bit unsigned integer, which I believe is still totally sufficient.

### CODE REVIEW

File: Spec.md [hunk 1]
```diff
@@ -1,124 +1,136 @@
...
 
 ## Token Fields
 
 ### Version
 
-This field denotes which version of the format is being used by
-the token. Currently there is only one version defined, with the
-value 128 (0x80).
+This field denotes which version of the format is being used by the token.
+Currently there is only one version defined, with the value 128 (0x80).
```
Comment (lbragstad) [inline Spec.md]: This commit is adding a second version.

File: Spec.md [hunk 2]
```diff
@@ -1,124 +1,136 @@
...
 
-Given a key and token, to verify that the token is valid and
-recover the original message, perform the following steps, in
-order:
+Given a key and token, to verify that the token is valid and recover the
+original message, perform the following steps, in order:
 
 1. base64url decode the token.
+
 2. Ensure the first byte of the token is 0x80.
```
Comment (lbragstad) [inline Spec.md]: This will need a `Change in` section amended to it as well, right?

Comment (dolph) [inline Spec.md]: Yes, definitely! Nice catch.

File: Spec.md (hunk 1 above)
Comment (dolph) [inline Spec.md]: Nice catch, I totally missed this as I was focused on the timestamp field. Will revise.

**Comment by dolph** [comment]
Test cases with microsecond data are in `generate.json`, `verify.json`, and `invalid.json`. For example:

  https://github.com/dolph/fernet-spec/blob/2688de4d318bda3fcfa2a4808418a06948c9039a/generate.json

I'm not sure what you mean by the "or do we wait ..." (this is already a pull request against fernet/spec).

And no, I haven't written any sort of real implementation yet.

**Comment by dolph** [comment]
Closing and abandoning this, as I'm no longer a stakeholder.

**Pull request closed without merging**